variables.tf

```diff
@@ -66,6 +66,16 @@ variable "autoclass" {
   default     = {}
 }
 
+variable "terminal_autoclass" {
+  description = "Optional map of storage class that objects in the bucket eventually transitions to. Supported values include: NEARLINE, ARCHIVE. Only used if autoclass is set as true. bucket_name => value, defaults to NEARLINE"
+  type        = map(string)
+  default     = {}
+  validation {
+    condition     = alltrue([for _, v in var.terminal_autoclass : var.autoclass == false || v == "NEARLINE" || v == "ARCHIVE"])
```

Comment:
Since we maintain a min TF compat with tf 1.3 we cannot use cross variable reference in validation rules (i.e checking var.autoclass) which was introduced in 1.9. If there is a validation error if this is set and  autoclass is disabled could we perform a check before passing value to terminal_storage_class - something like `lookup(var.autoclass..) ? lookup(var.terminal_autoclass...) : null`